src/api/request-hooks/hook.js

```diff
@@ -0,0 +1,39 @@
+import { RequestFilterRule } from 'testcafe-hammerhead';
+
+export default class RequestHook {
+    constructor (requestFilterRules, responseEventConfigureOpts) {
+        this.requestFilterRules              = this._prepareRequestFilterRules(requestFilterRules);
+        this._instantiatedRequestFilterRules = [];
+        this.responseEventConfigureOpts      = responseEventConfigureOpts;
+    }
+
+    _prepareRequestFilterRules (rules) {
+        if (rules)
+            return Array.isArray(rules) ? rules : [rules];
+
+        return [RequestFilterRule.ANY];
+    }
+
+    _instantiateRequestFilterRules () {
+        this.requestFilterRules.forEach(rule => {
+            if (rule instanceof RequestFilterRule)
+                this._instantiatedRequestFilterRules.push(rule);
+            else
+                this._instantiatedRequestFilterRules.push(new RequestFilterRule(rule));
+        });
+    }
+
+    onRequest (/*RequestEvent event*/) {
```

Comment:
If this class is abstract this method should throw an error. If a subclass shouldn't do anything here they just should have an empty method